rad/src/lib.rs

```diff
@@ -298,13 +301,21 @@ async fn http_response(
 
     // If at some point we want to support the retrieval of non-UTF8 data (e.g. raw bytes), this is
     // where we need to decide how to read the response body
-    let (_parts, mut body) = response.into_parts();
     let mut response_string = String::default();
-    body.read_to_string(&mut response_string)
-        .await
-        .map_err(|x| RadError::HttpOther {
-            message: x.to_string(),
-        })?;
+
+    let (parts, mut body) = response.into_parts();
+    match retrieve.kind {
+        RADType::HttpHead => {
+            response_string = format!("{:?}", parts.headers);
```

Comment:
Because we are relying on `impl Display for http::header::HeaderMap` here, I think we could use a test to ensure that a future version of the `http` library with a different `impl` doesn't break consensus.

Reply:
Forget about that! You made one down below! :rocket: